### PR TITLE
mgr/dashboard: Replace default dashboard with dashboard v2

### PR DESCRIPTION
**Pull request by LenzGr**
As [discussed on ceph-devel](https://www.spinics.net/lists/ceph-devel/msg40132.html), the initial Dashboard v2 pull request will not fully remove the old dashboard, but move it aside first, for later removal.

This pull requests moves the old dashboard code into the directory `dashboard_legacy` and moves the dashboard v2 code into its place instead. It also updates a few references to the previous directory name (`dashboard_v2`).

### CODE REVIEW

File: src/pybind/mgr/dashboard/README.rst [hunk 1]
```diff
@@ -1,80 +1,77 @@
+Dashboard and Administration Module for Ceph Manager (aka "Dashboard v2")
+=========================================================================
 
-=========================
-ceph-mgr dashboard module
-=========================
+Overview
+--------
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard/README.rst]: Does it makes sense to move these formatting changes to a different PR, so that we can merge them right now?

**Review by sebastian-philipp — DISMISSED** [review]

**Comment by LenzGr** [comment]
I'm closing this one, as we will perform a different operation to replace the "old" dashboard. Instead of renaming the directory, we'll remove it in one commit before renaming `dashboard_v2` into `dashboard` in the following commit, followed by a commit that replaces all corresponding `dashboard_v2` references.
